src/eastasianwidthdata.c

```diff
@@ -1,324 +1,15 @@
-    {0x3248, 0x324F, EAW_A},
-    {0x3250, 0x4DBF, EAW_W},
-    {0x4E00, 0xA48C, EAW_W},
-    {0xA490, 0xA4C6, EAW_W},
-    {0xA960, 0xA97C, EAW_W},
-    {0xAC00, 0xD7A3, EAW_W},
-    {0xE000, 0xF8FF, EAW_A},
-    {0xF900, 0xFAFF, EAW_W},
-    {0xFE00, 0xFE0F, EAW_A},
-    {0xFE10, 0xFE19, EAW_W},
-    {0xFE30, 0xFE52, EAW_W},
-    {0xFE54, 0xFE66, EAW_W},
-    {0xFE68, 0xFE6B, EAW_W},
-    {0xFF01, 0xFF60, EAW_F},
-    {0xFF61, 0xFFBE, EAW_H},
-    {0xFFC2, 0xFFC7, EAW_H},
-    {0xFFCA, 0xFFCF, EAW_H},
-    {0xFFD2, 0xFFD7, EAW_H},
-    {0xFFDA, 0xFFDC, EAW_H},
-    {0xFFE0, 0xFFE6, EAW_F},
-    {0xFFE8, 0xFFEE, EAW_H},
-    {0xFFFD, 0xFFFD, EAW_A},
-    {0x16FE0, 0x16FE4, EAW_W},
-    {0x16FF0, 0x16FF1, EAW_W},
-    {0x17000, 0x187F7, EAW_W},
-    {0x18800, 0x18CD5, EAW_W},
-    {0x18D00, 0x18D08, EAW_W},
-    {0x1AFF0, 0x1AFF3, EAW_W},
-    {0x1AFF5, 0x1AFFB, EAW_W},
-    {0x1AFFD, 0x1AFFE, EAW_W},
-    {0x1B000, 0x1B122, EAW_W},
-    {0x1B132, 0x1B132, EAW_W},
-    {0x1B150, 0x1B152, EAW_W},
-    {0x1B155, 0x1B155, EAW_W},
-    {0x1B164, 0x1B167, EAW_W},
-    {0x1B170, 0x1B2FB, EAW_W},
-    {0x1F004, 0x1F004, EAW_W},
-    {0x1F0CF, 0x1F0CF, EAW_W},
-    {0x1F100, 0x1F10A, EAW_A},
-    {0x1F110, 0x1F12D, EAW_A},
-    {0x1F130, 0x1F169, EAW_A},
-    {0x1F170, 0x1F18D, EAW_A},
-    {0x1F18E, 0x1F18E, EAW_W},
-    {0x1F18F, 0x1F190, EAW_A},
-    {0x1F191, 0x1F19A, EAW_W},
-    {0x1F19B, 0x1F1AC, EAW_A},
-    {0x1F200, 0x1F202, EAW_W},
-    {0x1F210, 0x1F23B, EAW_W},
-    {0x1F240, 0x1F248, EAW_W},
-    {0x1F250, 0x1F251, EAW_W},
-    {0x1F260, 0x1F265, EAW_W},
-    {0x1F300, 0x1F320, EAW_W},
-    {0x1F32D, 0x1F335, EAW_W},
-    {0x1F337, 0x1F37C, EAW_W},
-    {0x1F37E, 0x1F393, EAW_W},
-    {0x1F3A0, 0x1F3CA, EAW_W},
-    {0x1F3CF, 0x1F3D3, EAW_W},
-    {0x1F3E0, 0x1F3F0, EAW_W},
-    {0x1F3F4, 0x1F3F4, EAW_W},
-    {0x1F3F8, 0x1F43E, EAW_W},
-    {0x1F440, 0x1F440, EAW_W},
-    {0x1F442, 0x1F4FC, EAW_W},
-    {0x1F4FF, 0x1F53D, EAW_W},
-    {0x1F54B, 0x1F54E, EAW_W},
-    {0x1F550, 0x1F567, EAW_W},
-    {0x1F57A, 0x1F57A, EAW_W},
-    {0x1F595, 0x1F596, EAW_W},
-    {0x1F5A4, 0x1F5A4, EAW_W},
-    {0x1F5FB, 0x1F64F, EAW_W},
-    {0x1F680, 0x1F6C5, EAW_W},
-    {0x1F6CC, 0x1F6CC, EAW_W},
-    {0x1F6D0, 0x1F6D2, EAW_W},
-    {0x1F6D5, 0x1F6D7, EAW_W},
-    {0x1F6DC, 0x1F6DF, EAW_W},
-    {0x1F6EB, 0x1F6EC, EAW_W},
-    {0x1F6F4, 0x1F6FC, EAW_W},
-    {0x1F7E0, 0x1F7EB, EAW_W},
-    {0x1F7F0, 0x1F7F0, EAW_W},
-    {0x1F90C, 0x1F93A, EAW_W},
-    {0x1F93C, 0x1F945, EAW_W},
-    {0x1F947, 0x1F9FF, EAW_W},
-    {0x1FA70, 0x1FA7C, EAW_W},
-    {0x1FA80, 0x1FA88, EAW_W},
-    {0x1FA90, 0x1FABD, EAW_W},
-    {0x1FABF, 0x1FAC5, EAW_W},
-    {0x1FACE, 0x1FADB, EAW_W},
-    {0x1FAE0, 0x1FAE8, EAW_W},
-    {0x1FAF0, 0x1FAF8, EAW_W},
-    {0x20000, 0x2FFFD, EAW_W},
-    {0x30000, 0x3FFFD, EAW_W},
-    {0xE0100, 0xE01EF, EAW_A},
-    {0xF0000, 0xFFFFD, EAW_A},
-    {0x100000, 0x10FFFD, EAW_A},
-};
+bool op_is_east_asian(utf32_t ch) {
```

Comment:
For compatibility and potential future use, we should probably keep the old data, and add your new function as an optimization. Its name could be `ub_is_east_asian`, with doc comments explaining `F`, `W`, and `H` count as the "is" case.

Reply:
Comments added.

Regarding compatibility, would you like keep the `eaNew` and `eaLast` for binary compatibility?

The data in this file probably shouldn't be used by users, so I removed them. If future need arise, they can be added back as an implementation detail.

The function is named `op_is_east_asian` because it isn't meant to be applied to _any_ char; it's only for those with `LBP_OP`.

Reply:
OK, then `lb_is_op_east_asian` might be appropriate. My key point is that we should prefix *private* symbols. See also other `ub_…` and `lb_…` symbols.

Reply:
> Regarding compatibility, would you like keep the `eaNew` and `eaLast` for binary compatibility?

No, I meant keeping the `eaw_prop` data and `ub_get_char_eaw_class`. I think they are potentially useful later.

Keeping `eaNew` and `eaLast` and not using them may also be a good idea.... I am open at this moment. If so, we need to comment out code, instead of removing them entirely.

Reply:
> OK, then `lb_is_op_east_asian` might be appropriate. My key point is that we should prefix _private_ symbols. See also other `ub_…` and `lb_…` symbols.

Thinking again, `ub_is_op_east_asian` might be a better name.

Reply:
> No, I meant keeping the `eaw_prop` data and `ub_get_char_eaw_class`. I think they are potentially useful later.

All right.

> Keeping `eaNew` and `eaLast` and not using them may also be a good idea.... I am open at this moment. If so, we need to comment out code, instead of removing them entirely.

I'm not sure about "comment out code". The field is kept for binary compatibility purpose, but the code is not.

Reply:
> I'm not sure about "comment out code". The field is kept for binary compatibility purpose, but the code is not.

I meant commenting out code like the following:

```cpp
         (lbpCtx->lbcNew == LBP_OP &&
          !(lbpCtx->eaNew == EAW_F || lbpCtx->eaNew == EAW_W ||
            lbpCtx->eaNew == EAW_H))) ||
```

But as I said, the binary compatibility may not be so important—I could just bump up the `version-info`. I am open about which option to take.

Reply:
I'm fine not keeping binary compatibility.